LayoutTests/fast/css/stroke-dashoffset-parsing.html

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+<body>
+<script src="../../resources/js-test.js"></script>
+<div id="target"></div>
+<script>
+description("Tests the parsing of the stroke-dashoffset CSS property.");
+
+const target = document.getElementById("target");
+
+// Initial value.
+shouldBeEqualToString("getComputedStyle(target).strokeDashoffset", "0px");
+
+// Valid length.
+evalAndLog("target.style.strokeDashoffset = '3px';");
+shouldBeEqualToString("target.style.strokeDashoffset", "3px");
+shouldBeEqualToString("getComputedStyle(target).strokeDashoffset", "3px");
+
+// Valid percentage.
+evalAndLog("target.style.strokeDashoffset = '10%';");
+shouldBeEqualToString("target.style.strokeDashoffset", "10%");
+shouldBeEqualToString("getComputedStyle(target).strokeDashoffset", "10%");
+
+// Valid number.
+evalAndLog("target.style.strokeDashoffset = '20';");
+shouldBeEqualToString("target.style.strokeDashoffset", "20"); // Gecko says 20%, Blink says 20.
+shouldBeEqualToString("getComputedStyle(target).strokeDashoffset", "20px");
+
+// Zero value.
+evalAndLog("target.style.strokeDashoffset = '0';");
+shouldBeTrue("target.style.strokeDashoffset == '0' || target.style.strokeDashoffset == '0px'"); // Gecko says 0px, Blink says 0.
```

Comment:
Why do we allow either? css-values-3 says:

> However, if a 0 could be parsed as either a &lt;number> or a [&lt;length>](https://drafts.csswg.org/css-values-3/#length-value) in a property (such as [line-height](https://drafts.csswg.org/css2/#propdef-line-height)), it must parse as a &lt;number>.

so the value must be interpreted as a &lt;number>. Unless there's some other spec text somewhere that allows for either serialization.

Reply:
Yes, I hesitated. The reason I allowed both is that WebKit currently doesn't get it right and I didn't fix this yet.
So if I check for 0, we will have a FAIL line in our tests (which I guess is OK).

The issue is that our code currently calls consumeLength() and then falls back to calling consumeNumber() if the earlier didn't work. For 0, consumeLength() currently happily succeeds and returns 0px.